code/game/objects/items.dm

```diff
@@ -342,10 +342,12 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
 			. += "[src] is made of cold-resistant materials."
 		if(resistance_flags & FIRE_PROOF)
 			. += "[src] is made of fire-retardant materials."
-
-	if(!user.research_scanner)
 		return
 
+/obj/item/examine_more(mob/user)
+	. = ..()
+	if(!HAS_TRAIT(user, TRAIT_RESEARCH_SCANNER))
+		return
 	/// Research prospects, including boostable nodes and point values. Deliver to a console to know whether the boosts have already been used.
 	var/list/research_msg = list("<font color='purple'>Research prospects:</font> ")
```

Comment:
Can you make this its own proc that is called if they *do* have the trait? It would help make this code a lot easier to read.

Reply:
I thiiink so? Check the latest version and tell me if that's what you were looking for.